src/common/hist_util.h

```diff
@@ -60,14 +60,14 @@ struct HistCutUnit {
   // default constructor
   HistCutUnit() {}
   // constructor
-  HistCutUnit(const bst_float* cut, unsigned size)
+  HistCutUnit(const bst_float* cut, size_t size)
       : cut(cut), size(size) {}
 };
 
 /*! \brief cut configuration for all the features */
 struct HistCutMatrix {
-  /*! \brief actual unit pointer */
-  std::vector<unsigned> row_ptr;
+  /*! \brief unit pointer to rows by element position */
+  std::vector<size_t> row_ptr;
```

Comment:
This change is unnecessary -- `HistCutMatrix` is meant to store quantile points for each feature. With a typical configuration of `max_bin=256`, it is unlikely that quantile points would exceed 2^{32} in number.